rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -624,32 +605,6 @@ func validateBatch(batchIndex uint64, event *L1FinalizeBatchEvent, parentFinaliz
 	return endBlock.Header.Number.Uint64(), finalizedBatchMeta, nil
 }
 
-// determineCodecVersion determines the codec version based on the block number and chain configuration.
-// If the codecVersion is not provided (nil), which can happen with older client versions,
-// it will be inferred from the hardfork rules.
-//
-// Note: The codecVersion (except genesis batch with version 0) is retrieved from the commit batch transaction calldata and stored in the database.
-// This function provides backward compatibility when the codecVersion is not available in the database,
-// which can occur with older client versions that don't store this information.
-func determineCodecVersion(startBlockNumber *big.Int, startBlockTimestamp uint64, chainCfg *params.ChainConfig, providedCodecVersion *uint8) encoding.CodecVersion {
-	if providedCodecVersion != nil {
-		return encoding.CodecVersion(*providedCodecVersion)
-	}
-
-	switch {
-	case startBlockNumber.Uint64() == 0 || !chainCfg.IsBernoulli(startBlockNumber):
-		return encoding.CodecV0 // codecv0: genesis batch or batches before Bernoulli
-	case !chainCfg.IsCurie(startBlockNumber):
-		return encoding.CodecV1 // codecv1: batches after Bernoulli and before Curie
-	case !chainCfg.IsDarwin(startBlockTimestamp):
-		return encoding.CodecV2 // codecv2: batches after Curie and before Darwin
-	case !chainCfg.IsDarwinV2(startBlockTimestamp):
-		return encoding.CodecV3 // codecv3: batches after Darwin
-	default:
-		return encoding.CodecV4 // codecv4: batches after DarwinV2
-	}
-}
-
 // decodeBlockRangesFromEncodedChunks decodes the provided chunks into a list of block ranges.
 func decodeBlockRangesFromEncodedChunks(codecVersion encoding.CodecVersion, chunks [][]byte) ([]*rawdb.ChunkBlockRange, error) {
```

Comment:
This function should belong to da-codec repo.

Reply:
put this function here looks more suitable IMO. because the function encapsulates initializing codec logic. and the return value is `[]*rawdb.ChunkBlockRange`, which is an l2geth's type.